DataFormats/MemoryResources/include/MemoryResources/MemoryResources.h

```diff
@@ -208,7 +208,7 @@ class SpectatorAllocator : public boost::container::pmr::polymorphic_allocator<T
   {
   }
 
-  T* allocate(size_t size) { return reinterpret_cast<T*>(this->resource()->allocate(size * sizeof(T), 0)); }
+  T* allocate(size_t size) { return reinterpret_cast<T*>(this->resource()->allocate(size * sizeof(T), 64)); }
   void deallocate(T* ptr, size_t size)
   {
     this->resource()->deallocate(const_cast<typename std::remove_cv<T>::type*>(ptr), size);
```

Comment:
There might be one point we have to consider:
C++ uses different allocators for aligned / non-aligned memory.
Therefore, when using new / delete or operator new / delete, one has to pass the alignment constraints also to delete.
This will make sure the correct deallocator is called.
@ktf  @rbx: Are we doing this at the moment?

Normally, this will not break when the mismatched deallocator is used, because by default they are the same.
But e.g. if someone overloads the unaligned new / delete but not the aligned ones, it could theoretically break.

Reply:
In FairMQ currently only the shared memory transport honors the alignment argument. And it passes that to boost via `boost::interprocess::managed_shared_memory::allocate_aligned(size, alignment)`. In their docs the corresponding deallocation method to has no alignment argument, nor is one available in the API. So I assume it does the right thing.

Reply:
OK, but then I guess you should still foresee the alignment parameter in the deallocate function, such that user can pass it already.
This will then allow you to extent aligned alloc/dealloc to other allocators, which require the match.